tests/lib/utils/vue-component.js

```diff
@@ -0,0 +1,257 @@
+/**
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../../../lib/utils/index')
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const rule = {
+  create (context) {
+    return utils.executeOnVueComponent(context, obj => {
+      context.report({
+        node: obj,
+        message: 'Internal test: component.'
+      })
+    })
+  },
+  meta: {
+    fixable: null,
+    schema: []
+  }
+}
+
+const RuleTester = require('eslint').RuleTester
+const parserOptions = {
+  ecmaVersion: 6,
+  sourceType: 'module'
+}
+
+function makeError (line) {
+  return {
+    message: 'Internal test: component.',
+    line,
+    type: 'ObjectExpression'
+  }
+}
+
+function notFoundTests (ext) {
+  return [
+    {
+      filename: `test.${ext}`,
+      code: `export const foo = {}`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `export var foo = {}`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `const foo = {}`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `var foo = {}`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `let foo = {}`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `foo({ })`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `foo(() => { return {} })`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `Vue.component('async-example', function (resolve, reject) { })`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `Vue.component('async-example', function (resolve, reject) { resolve({}) })`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `new Vue({ })`,
+      parserOptions
+    },
+    {
+      filename: `test.${ext}`,
+      code: `{
+        foo: {}
+      }`,
+      parserOptions
+    }
+  ]
+}
+
+function foundTests (ext) {
+  return [
+    {
+      filename: `test.${ext}`,
+      code: `// ${ext}
+      Vue.component('async-example', function (resolve, reject) {
+        // @vue/component
+        resolve({})
+      })`,
+      parserOptions,
+      errors: [makeError(4)]
+    },
+    {
+      filename: `test.${ext}`,
+      code: `Vue.component({})`,
+      parserOptions,
+      errors: [makeError(1)]
+    },
+    {
+      filename: `test.${ext}`,
+      code: `// ${ext}
+      // @vue/component
+      export default { }`,
+      parserOptions,
+      errors: [makeError(3)]
+    },
+    {
+      filename: `test.${ext}`,
+      code: `// ${ext}
```

Comment:
Please consider formatting those code example as follows:
```
code: `
  /* @vue/component */
  export default {}
`
```
I think they would be much easier to read :)